### PR TITLE
Await turnOn if not completed + Throw when turnOn has not been called yet

### LINKED ISSUE #14: App stuck because Computer is not turned on
We've integrated computer in our app and on iOS the app is stuck on the splash screen. After some looking the issue seems to be that the computer hasn't started yet.
```dart
  Future<Map<String, dynamic>> decode(String encoded) async {
    if (!computer.isRunning) {
      print('computer isn\'t turned on yet'); // <-- This is printed
    }
    return computer.compute<String, Map<String, dynamic>>(parseAndDecode, param: encoded);
  }
```
I would expect the package to either wait in the compute function until the computer is available or, if that's to hard, throw an exception so that we know why the app stopped functioning. Now it just waits forever.

### PR DESCRIPTION
**Pull request by ikbendewilliam**
Fixes #14 
Fixes typo `workersCount` should *not* be more than 0 -> `workersCount` should be more than 0

### CODE REVIEW

File: lib/src/computer.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@ import 'compute_api/compute_api.dart';
 /// Class, that provides `compute` like API for concurrent calculations
 class Computer {
   final _computeDelegate = ComputeAPI();
+  Completer? _completer;
```
Comment (NicolaVerbeeck) [inline lib/src/computer.dart:8]: Does not describe what the completer is for

File: lib/src/computer.dart [hunk 2]
```diff
@@ -17,29 +18,37 @@ class Computer {
...
 
-  /// Turn on `Computer`, `workersCount` should not be more than 0, default is 2
+  /// Turn on `Computer`, `workersCount` should be more than 0, default is 2
   /// `verbose` is false by default, enabling it leads to logging of every operation
   Future<void> turnOn({
     int workersCount = 2,
     bool verbose = false,
   }) async {
-    return _computeDelegate.turnOn(
+    _completer = Completer<void>();
```
Comment (NicolaVerbeeck) [inline lib/src/computer.dart:27]: add assert that checks > 0

Comment (NicolaVerbeeck) [reply lib/src/computer.dart:27]: check if we are not turning on twice

File: lib/src/computer.dart [hunk 3]
```diff
@@ -17,29 +18,37 @@ class Computer {
...
+    await _computeDelegate.turnOn(
       workersCount: workersCount,
       verbose: verbose,
     );
+    _completer?.complete();
   }
 
   /// Executes function `fn` with passed `param`. Takes only top-level functions and static methods.
   /// `P` is `param` type, `R` is function return type
+  /// Throws when turnOn has not been called, awaits when turnOn has been called, but not yet finished
```
Comment (NicolaVerbeeck) [inline lib/src/computer.dart:37]: Update wording `awaits when turnOn has been called`

File: lib/src/computer.dart [hunk 4]
```diff
@@ -17,29 +18,37 @@ class Computer {
...
+    if (_completer == null) {
+      throw StateError('Computer is not running, you must run turnOn first!');
+    }
+    await _completer!.future;
     return _computeDelegate.compute<P, R>(fn, param: param);
   }
 
   /// Turn off `Computer`
   Future<void> turnOff() async {
+    if (_completer?.isCompleted == false) _completer!.completeError('Computer turned off');
```
Comment (NicolaVerbeeck) [inline lib/src/computer.dart:51]: Turn off does not reset completer

**Review by NicolaVerbeeck — CHANGES_REQUESTED** [review]

File: lib/src/computer.dart [hunk 5]
```diff
@@ -17,29 +20,42 @@ class Computer {
...
+    await _completer!.future;
     return _computeDelegate.compute<P, R>(fn, param: param);
   }
 
   /// Turn off `Computer`
   Future<void> turnOff() async {
+    if (_completer?.isCompleted == false) {
+      _completer!.completeError('Computer turned off');
+    }
+    _completer = null;
```
Comment (NicolaVerbeeck) [inline lib/src/computer.dart:58]: Should we wait for turn off to complete before marking this. and/or wait for turn on to complete if it was running at that time

**Comment by ikbendewilliam** [comment]
@dmitriismitnov @91team can you review please?

**Comment by dmitriismitnov** [comment]
@ikbendewilliam Hi =) I will review this pr 04th January 2021. Happy New Year 🎆

**Comment by dmitriismitnov** [comment]
@ikbendewilliam Oh, thanks for your pr!

**Comment by ikbendewilliam** [comment]
@dmitriismitnov great! Thank you! Happy new year 🥳

**Comment by dmitriismitnov** [comment]
@ikbendewilliam Hi. Unfortunately, I cannot approve this PR.

This PR conflicts with https://github.com/91team/computer/pull/12 therefore the tests are not valid.
For more details, please pay attention to https://github.com/91team/computer/pull/12

In Future, for resolve this collision I will create some mechanism for configure various behaviour.
If you want to be involved, welcome =)

**Comment by ikbendewilliam** [comment]
I see, #12 was merged after I started on this. I'm not sure if that PR fixes all issues, I'll be checking that when I've got the time 👍

**Pull request closed without merging**